pkgs/os-specific/linux/keyutils/default.nix

```diff
@@ -20,6 +20,12 @@ stdenv.mkDerivation rec {
     # Before removing this patch, please ensure the package still builds by running eg.
     # nix-build -E 'with import ./. {}; pkgs.keyutils.override { stdenv = pkgs.llvmPackages_latest.stdenv; }'
     ./0001-Remove-unused-function-after_eq.patch
+
+    # Fix build for s390-linux, where size_t is different from ptrdiff_t.
+    (fetchurl {
```

Comment:
Did you try using fetchpatch?

Reply:
1. https://github.com/NixOS/nixpkgs/blob/d8a62e205d4c8f88147bd18068a65b5eba5364e4/pkgs/os-specific/linux/keyutils/default.nix#L3-L6
2. It's better to use `fetchurl` instead of `fetchpatch` for patches that aren't generated, and so won't change under us.  It has fewer dependencies, and it doesn't depend on specifics of a single version of patchutils.